### DISCUSSION TITLE
The product file doesn't give you the option to also set the name of the repositories that your product build on (updates from)

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-equinox/p2#342] [General]
Hi,

not sure where the product file belongs to that makes the product (is this platform or rcp? or p2?)

So i asked the question here, in the product file (editor) you can define your repositories where this product builds from (or updates from)

Problem is the name is not asked for, and then when you install it, in the preferences it is displayed very weird, because you have a lot without a name. The name seems kind of a bit mandatory (also in the p2.inf where i now place it)

Is there a reason for that? i tries to just adjust the product xml file by adding "name=xxx" or "displayname=xxx" but that isn't picked up.

**Comment by laeubi** [comment]
The product file itself is defined by pde:

- https://github.com/eclipse-pde/eclipse.pde

p2 is just reading this.

**Comment by mickaelistria** [comment]
> Is there a reason for that?

I don't think there is a reason for that, nor a reason against that. So if you wish to improve the .product definition so one can set names to repo, and to improve the .product editor to make it easy to configure and improve the p2 task that publishes a product so this metadata remains at deployment and improve the p2 task that installs a product so that this new metadata is read at installation to configure the repo name, that would be welcome ;)

**Reply by jcompagner** [reply]
i guess a lot of stuff should already be there right? because with the "p2.inf" i do provide the repository info already (and that is used correctly as far as i can see)

**Reply by mickaelistria** [reply]
Probably.

**Comment by HannesWell** [comment]
> i guess a lot of stuff should already be there right? because with the "p2.inf" i do provide the repository info already (and that is used correctly as far as i can see)

That's right. Coincidentially I looked this a few days ago and think it should be added.

And yes P2 processes the `repository` elements in the `ProductFile` class. There you can enhance the generated p2-instructions to also concider a `name` element.
PDE provides the UI to manipulate the repo-refs in the UpdatesPage of the product Editor (not sure about the exact name, but it is next to the Product's `OverviewPage`)

**Comment by HannesWell** [comment]
Created https://github.com/eclipse-equinox/p2/issues/345 to track the work on this issue?
